packages/medusa/src/services/line-item-adjustment.ts

```diff
@@ -0,0 +1,226 @@
+   */
+  async retrieve(
+    id: string,
+    config: FindConfig<LineItemAdjustment> = {}
+  ): Promise<LineItemAdjustment> {
+    const lineItemAdjustmentRepo = this.manager_.getCustomRepository(
+      this.lineItemAdjustmentRepo_
+    )
+
+    const query = this.buildQuery_({ id }, config)
+    const lineItemAdjustment = await lineItemAdjustmentRepo.findOne(query)
+
+    if (!lineItemAdjustment) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `Line item adjustment with id: ${id} was not found`
+      )
+    }
+
+    return lineItemAdjustment
+  }
+
+  /**
+   * Creates a line item adjustment
+   * @param data - the line item adjustment to create
+   * @return line item adjustment
+   */
+  async create(data: Partial<LineItemAdjustment>): Promise<LineItemAdjustment> {
+    return await this.atomicPhase_(async (manager: EntityManager) => {
+      const lineItemAdjustmentRepo = manager.getCustomRepository(
+        this.lineItemAdjustmentRepo_
+      )
+
+      const lineItemAdjustment = await lineItemAdjustmentRepo.create(data)
+
+      return await lineItemAdjustmentRepo.save(lineItemAdjustment)
+    })
+  }
+
+  /**
+   * Creates a line item adjustment
+   * @param id - the line item adjustment id to update
+   * @param data - the line item adjustment to create
+   * @return line item adjustment
+   */
+  async update(
+    id: string,
+    data: Partial<LineItemAdjustment>
+  ): Promise<LineItemAdjustment> {
+    return await this.atomicPhase_(async (manager: EntityManager) => {
+      const lineItemAdjustmentRepo = manager.getCustomRepository(
+        this.lineItemAdjustmentRepo_
+      )
+
+      const lineItemAdjustment = await this.retrieve(id)
+
+      const { metadata, ...rest } = data
+
+      if (metadata) {
+        lineItemAdjustment.metadata = this.setMetadata_(
+          lineItemAdjustment,
+          metadata
+        )
+      }
+
+      for (const [key, value] of Object.entries(rest)) {
+        lineItemAdjustment[key] = value
+      }
+
+      const result = await lineItemAdjustmentRepo.save(lineItemAdjustment)
+      return result
+    })
+  }
+
+  /**
+   * Lists line item adjustments
+   * @param selector - the query object for find
+   * @param config - the config to be used for find
+   * @return the result of the find operation
+   */
+  async list(
+    selector: FilterableLineItemAdjustmentProps = {},
+    config: FindConfig<LineItemAdjustment> = { skip: 0, take: 20 }
+  ): Promise<LineItemAdjustment[]> {
+    const lineItemAdjustmentRepo = this.manager_.getCustomRepository(
+      this.lineItemAdjustmentRepo_
+    )
+
+    const query = this.buildQuery_(selector, config)
+    return await lineItemAdjustmentRepo.find(query)
+  }
+
+  /**
+   * Deletes line item adjustments matching a selector
+   * @param selector - the query object for find
+   * @return the result of the delete operation
+   */
+  async delete(selector: FilterableLineItemAdjustmentProps): Promise<void> {
```

Comment:
hmm .. not sure how to do the overload without doing something like `async delete(selectorOrId: string | FilterableLineItemAdjustmentProps)` which is a bit awkward imo. However, I agree that we should try to be consistent. I think it might be a good idea to leave this as it is right now while we're still making adjustments to existing services. As of now, I've needed to delete one or many adjustments either by line item id, discount id, or id (but there might be more). Might then create a method for each of these like this `deleteAdjustmentsByLineItem` , `deleteAdjustmentsByDiscount`, and then a plain `delete` etc. Let me know what you think!